File: exp/UnityChan2DTutorial/Assets/ScriptsCSharp/UI/ChangeScore.cs
using UnityEngine;
using UnityEngine.UI;
using System.Collections;

public class ChangeScore : MonoBehaviour {

	void ScoreUpdate(){
		int amount = GameObject.FindWithTag("Player").GetComponent<PlayerState>().score ;
		string number = amount + "" ;
		string zero = "000000000" ;

		GetComponent<Text>().text =  zero.Substring(0,zero.Length - number.Length) + amount ;
	}
}

File: exp/UnityChan2DTutorial/Assets/ScriptsJs/UI/ChangeScore.js
#pragma strict

function ScoreUpdate () {
  var amount : int = gameObject.FindWithTag("Player").GetComponent.<PlayerState>().score ;
  var number : String = amount+"" ; 
  var zero : String = "000000000" ;
  
  GetComponent.<UI.Text>().text =  zero.Substring(0,zero.length-number.length) + amount ;
}